plugins/robots/generators/nxt/nxtOsekCGenerator/nxtOsekCGeneratorPlugin.cpp

```diff
@@ -223,21 +229,31 @@ void NxtOsekCGeneratorPlugin::uploadProgram()
 void NxtOsekCGeneratorPlugin::checkNxtTools()
 {
 	const QDir dir(PlatformInfo::invariantSettingsPath("pathToNxtTools"));
-	if (!dir.exists()) {
+	const QDir gnuarm(dir.absolutePath() + "/gnuarm");
+	const QDir nexttool(dir.absolutePath() + "/nexttool");
+	const QDir nxtOSEK(dir.absolutePath() + "/nxtOSEK");
+
+	if (!dir.exists() || !gnuarm.exists() || !nexttool.exists() || !nxtOSEK.exists()) {
 		mNxtToolsPresent = false;
+		QLOG_ERROR() << "Missing" << dir.absolutePath() << "or mandatory subdirectory" <<
+			dir.entryList(QDir::Filter::NoFilter, QDir::SortFlag::DirsFirst | QDir::SortFlag::Name);
 	} else {
-		QDir gnuarm(dir.absolutePath() + "/gnuarm/bin");
-		QDir nexttool(dir.absolutePath() + "/nexttool");
-		QDir nxtOSEK(dir.absolutePath() + "/nxtOSEK");
-
-#ifdef Q_OS_WIN
-		QFile compile1(dir.absolutePath() + "/compile.bat");
-		QFile compile2(dir.absolutePath() + "/compile.sh");
-		mNxtToolsPresent = gnuarm.exists() && nexttool.exists() && nxtOSEK.exists()
-				&& compile1.exists() && compile2.exists();
-#else
-		QFile compile(dir.absolutePath() + "/compile.sh");
-		mNxtToolsPresent = gnuarm.exists() && nexttool.exists() && nxtOSEK.exists() && compile.exists();
-#endif
+		switch(QOperatingSystemVersion::currentType()) {
```

Comment:
I suggest using `PlatformInfo::osType` or introducing a companion function to that one to return some useful enumerable type.